pkg/ingress/kube/configmap/mcp_server.go

```diff
@@ -381,17 +387,36 @@ func (m *McpServerController) ConstructEnvoyFilters() ([]*config.Config, error)
 func (m *McpServerController) constructMcpSessionStruct(mcp *McpServer) string {
 	// Build match_list configuration
 	matchList := "[]"
+	var matchConfigs []string
 	if len(mcp.MatchList) > 0 {
-		matchConfigs := make([]string, len(mcp.MatchList))
-		for i, rule := range mcp.MatchList {
-			matchConfigs[i] = fmt.Sprintf(`{
+		for _, rule := range mcp.MatchList {
+			matchConfigs = append(matchConfigs, fmt.Sprintf(`{
 				"match_rule_domain": "%s",
 				"match_rule_path": "%s",
 				"match_rule_type": "%s"
-			}`, rule.MatchRuleDomain, rule.MatchRulePath, rule.MatchRuleType)
+			}`, rule.MatchRuleDomain, rule.MatchRulePath, rule.MatchRuleType))
+		}
+	}
+
+	if m.reconclier != nil {
+		vsFromMcp := m.reconclier.GetAllConfigs(gvk.VirtualService)
+		for _, c := range vsFromMcp {
+			vs := c.Spec.(*networking.VirtualService)
+			var host string
+			if len(vs.Hosts) > 1 {
+				host = fmt.Sprintf("^(%s)$", strings.Join(vs.Hosts, "|"))
```

Comment:
This has already been set up:
https:github.com/alibaba/higress/blob/7955aec63983ca7921edbc08ccecc83dab99ee36/plugins/golang-filter/mcp-session/common/match.go#L50